web/src/main/kotlin/com/example/ajaxproject/nats/GetUserByIdNatsController.kt

```diff
@@ -1,38 +1,35 @@
 package com.example.ajaxproject.nats
 
-import com.example.ajaxproject.GetByIdUserRequest
-import com.example.ajaxproject.GetByIdUserResponse
 import com.example.ajaxproject.NatsSubject.GET_USER_BY_ID
-import com.example.ajaxproject.User
-import com.example.ajaxproject.model.toProtoUser
+import com.example.ajaxproject.UserOuterClass.GetByIdUserRequest
+import com.example.ajaxproject.UserOuterClass.GetByIdUserResponse
 import com.example.ajaxproject.service.interfaces.UserService
 import com.google.protobuf.Parser
 import io.nats.client.Connection
 import org.springframework.stereotype.Component
+import reactor.core.publisher.Mono
 
 @Component
 class GetUserByIdNatsController(
-    override val connection: Connection,
+    private val userMapper: UserMapper,
     private val userService: UserService,
+    override val connection: Connection,
 ) : NatsController<GetByIdUserRequest, GetByIdUserResponse> {
 
-    override val subject = GET_USER_BY_ID
-    override val parser: Parser<GetByIdUserRequest> = GetByIdUserRequest.parser()
+    override val subject: String = GET_USER_BY_ID
 
-    override fun handle(request: GetByIdUserRequest): GetByIdUserResponse = runCatching {
-        val getUserById = userService.getUserById(request.userId)
-        buildSuccessResponse(getUserById.toProtoUser())
-    }.getOrElse { exception ->
-        buildFailureResponse(exception.javaClass.simpleName, exception.toString())
-    }
+    override val parser: Parser<GetByIdUserRequest> = GetByIdUserRequest.parser()
 
-    private fun buildSuccessResponse(user: User): GetByIdUserResponse =
-        GetByIdUserResponse.newBuilder().apply {
-            successBuilder.setUser(user)
-        }.build()
+    override fun generateReplyForNatsRequest(request: GetByIdUserRequest): Mono<GetByIdUserResponse> {
+        val parkingId = request.userId
 
-    private fun buildFailureResponse(exception: String, message: String): GetByIdUserResponse =
-        GetByIdUserResponse.newBuilder().apply {
-            failureBuilder.setMessage("User find by id failed by $exception: $message")
-        }.build()
+        return userService.getById(parkingId)
+            .map { userMapper.userToProto(it) }
+            .map {
```

Comment:
```suggestion
            .map {
```
Let's merge two `map` into one?